NitroxClient/GameLogic/Spawning/PrefabChildEntitySpawner.cs

```diff
@@ -8,24 +9,26 @@ namespace NitroxClient.GameLogic.Spawning
 {
     public class PrefabChildEntitySpawner : EntitySpawner<PrefabChildEntity>
     {
-        // When we first encounter a PrefabChildEntity, we simply need to assign it the right id matching the server
+        // When we encounter a PrefabChildEntity, we need to assign the id to a prefab with the same class id and index.
         public override IEnumerator SpawnAsync(PrefabChildEntity entity, TaskResult<Optional<GameObject>> result)
         {
             GameObject parent = NitroxEntity.RequireObjectFrom(entity.ParentId);
+            PrefabIdentifier prefab = parent.GetAllComponentsInChildren<PrefabIdentifier>()
+                                            .Where(prefab => prefab.classId == entity.ClassId)
+                                            .Skip(entity.ComponentIndex)
+                                            .First();
```

Comment:
Is there a reason you didn't choose `ElementAt()`? Seems more intuitive to me.

Reply:
makes sense.